### PR TITLE
fix: add 'JNI_RESULT_SET_NULL' error code

### PR DESCRIPTION
**Pull request by huolibo**

### CODE REVIEW

File: source/client/src/clientJniConnector.c [hunk 1]
```diff
@@ -489,7 +488,7 @@ JNIEXPORT jint JNICALL Java_com_taosdata_jdbc_TSDBJNIConnector_fetchRowImp(JNIEn
                numOfFields);
       return JNI_FETCH_END;
     } else {
-      jniDebug("jobj:%p, conn:%p, interrupted query", jobj, tscon);
+      jniDebug("jobj:%p, conn:%p, interrupted query. error code: %d", jobj, tscon, code);
```
Comment (sangshuduo) [inline source/client/src/clientJniConnector.c]: can we print the error reason returned by taos_errstr() too?

File: source/client/src/clientJniConnector.c [hunk 2]
```diff
@@ -584,7 +583,7 @@ JNIEXPORT jint JNICALL Java_com_taosdata_jdbc_TSDBJNIConnector_fetchBlockImp(JNI
       jniDebug("jobj:%p, conn:%p, resultset:%p, no data to retrieve", jobj, tscon, (void *)res);
       return JNI_FETCH_END;
     } else {
-      jniError("jobj:%p, conn:%p, query interrupted", jobj, tscon);
+      jniError("jobj:%p, conn:%p, query interrupted. error code: %d", jobj, tscon, error_code);
```
Comment (sangshuduo) [inline source/client/src/clientJniConnector.c]: same here

File: source/client/src/clientTmqConnector.c [hunk 3]
```diff
@@ -303,7 +303,7 @@ JNIEXPORT jint JNICALL Java_com_taosdata_jdbc_tmq_TMQConnector_fetchRawBlockImp(
       jniDebug("jobj:%p, conn:%p, resultset:%p, no data to retrieve", jobj, tscon, (void *)res);
       return JNI_FETCH_END;
     } else {
-      jniError("jobj:%p, conn:%p, query interrupted", jobj, tscon);
+      jniError("jobj:%p, conn:%p, query interrupted, error code: %d", jobj, tscon, error_code);
```
Comment (sangshuduo) [inline source/client/src/clientTmqConnector.c]: same here

**Review by sangshuduo — CHANGES_REQUESTED** [review]
see my comments

File: source/client/src/clientJniConnector.c (hunk 1 above)
Comment (huolibo) [reply source/client/src/clientJniConnector.c]: error message has been added

**Review by sangshuduo — APPROVED** [review]
LGTM

**Pull request merged**